azure/scope/cluster.go

```diff
@@ -382,6 +425,19 @@ func (s *ClusterScope) NodeRouteTable() infrav1.RouteTable {
 	return subnet.RouteTable
 }
 
+// NodeNatGateway returns the cluster node nat gateway.
+func (s *ClusterScope) NodeNatGateway() infrav1.NatGateway {
+	subnet, _ := s.AzureCluster.Spec.NetworkSpec.GetNodeSubnet()
+	return subnet.NatGateway
+}
+
+// SetNodeNatGateway sets the node subnet into the scope with the updated nat gateway.
+func (s *ClusterScope) SetNodeNatGateway(natGateway infrav1.NatGateway) {
+	nodeSubnet := s.NodeSubnet()
+	nodeSubnet.NatGateway = natGateway
+	s.AzureCluster.Spec.NetworkSpec.UpdateNodeSubnet(nodeSubnet)
```

Comment:
what's the difference between this and SetSubnet added in https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/4cff58a1e31902f79dedff1b7ff8332af3e67997 ? can we consolidate the two if they are the same?

Reply:
I changed it to use the new function.